cmd/soroban-rpc/internal/preflight/preflight.go

```diff
@@ -174,10 +174,11 @@ func getLedgerInfo(params PreflightParameters) (C.ledger_info_t, error) {
 	li := C.ledger_info_t{
 		network_passphrase: C.CString(params.NetworkPassphrase),
 		sequence_number:    C.uint32_t(simulationLedgerSeq),
-		protocol_version:   20,
+		protocol_version:   21,
```

Comment:
It would be nice to cover two protocol versions in the test suite: the 'current' one (20 in our case) and the 'next' one (21 in our case). The reason is that there is some period of time where a new version of RPC will still run for p20. That's not necessary for this PR (probably better to keep it concise), but just something to do as a followup for the additional test coverage.

Reply:
Yes, and we shouldn't hardcode the version in the simulation library https://github.com/stellar/soroban-rpc/issues/137

Reply:
https://github.com/stellar/soroban-rpc/issues/138

Reply:
Yes, env itself has to support all the protocol versions. We also have the recording mode/simulation coverage for both protocols 20 and 21 and the plan is for simulation library to always cover the curr/next protocol in the way that I've described in my initial comment.

Reply:
FWIW Core will run two versions of env during the transition period, but that's for reducing the risks of divergence; I don't think this is relevant/necessary for the simulation.

Reply:
Awesome, thanks!